### DISCUSSION TITLE
Get unset relations

### DISCUSSION THREAD

**Discussion by Fractal-Tess** [pocketbase/pocketbase#2444] [Q&A] [answered]
Hey there,

I have just a quick question.
How do you query records that do not have an active relationship set?
I have the following collection:
![image](https://user-images.githubusercontent.com/75957529/236877125-138d113d-8c4f-4517-af99-6549f9d54665.png)
And my goal is to query for any records in `video_collection_v2` that do not have a `parent_collection` entry.

The filter `parent_collection=null` seems to work but for only a single relation-based collection.
Also, while we are on that topic, how exactly do you query for entries that do have a set relation?

Thanks for your development of pocketbase <3

Cheers, 
Fractal-Tess

**Comment by ganigeorgiev** [comment, marked as answer]
> How do you query records that do not have an active relationship set?

You can use the [`:length` field modifier](https://pocketbase.io/docs/api-rules-and-filters/#length-modifier) to check the relation elements count.
`null` or empty string doesn't work with `multiple` relation fields because internally multiple valued fields are stored as serialized json array, aka. the empty value would be `'[]'`.

```js
parent_collection:length = 0
// or compare directly the raw serialized value
parent_collection = "[]"
```

> Also, while we are on that topic, how exactly do you query for entries that do have a set relation?

I'm not sure that I understand what do you mean.
If you are asking for the reverse of the above you can use negation (aka. `!=`) with the condition.

If you are asking for how to apply a filter condition on some of the relation fields, you can use dot-notation:

```js
// this will require EVERY related item to have a "someField" equal to 10
parent_collection.someField = 10

// if you want ANY/AT LEAST ONE OF type of condition you can prefix the operator with ?
parent_collection.someField ?= 10
```

If you are asking for how to expand relations, you can check https://pocketbase.io/docs/expanding-relations/. With the JS SDK that could look something like:

```js
await pb.collection("video_collection_v2").getList(1, 20, {
    expand: "parent_collection"
})
```

**Reply by Fractal-Tess** [reply]
Well, I was interested in query filtering records that have at least 1 or more relations, but as you mentioned the `:length` field modifier works wonders for that with the `>/<` operators.

Thanks a lot for your response <3
Take care!
